casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -385,4 +338,51 @@ object MultiParentCasperImpl {
 
     Log[F].info(s"Validating block ${PrettyPrinter.buildString(b, short = true)}.") *> validationProcessDiag
   }
+
+  def lastFinalizedBlock[F[_]: Sync: BlockDagStorage: BlockStore]: F[BlockMessage] =
+    for {
+      dag          <- BlockDagStorage[F].getRepresentation
+      blockMessage <- dag.lastFinalizedBlockUnsafe.flatMap(BlockStore[F].getUnsafe)
+    } yield blockMessage
+
+  def handleValidBlock[F[_]: Sync: BlockDagStorage: BlockStore: CasperBufferStorage](
+      block: BlockMessage
+  ): F[DagRepresentation] =
+    for {
+      updatedDag <- BlockDagStorage[F].insert(block, invalid = false)
+      _          <- CasperBufferStorage[F].remove(block.blockHash)
+      _          <- lastFinalizedBlock
```

Comment:
This were supposed to update finalization (thats why output is discarded), but now it looks like it is just reading of a value. It's fine for now as finalization is to be implemented, but just to want to point at this.

Reply:
This was previous code. Do you think it's different?
```scala
private def updateLastFinalizedBlock(newBlock: BlockMessage): F[Unit] =
  lastFinalizedBlock.whenA(
    newBlock.body.state.blockNumber % casperShardConf.finalizationRate == 0
  )

def lastFinalizedBlock: F[BlockMessage] =
  for {
    dag          <- blockDag
    blockMessage <- dag.lastFinalizedBlockUnsafe.flatMap(BlockStore[F].getUnsafe)
  } yield blockMessage
```

Reply:
After recent PRs with removal of estimator and pure DagRepresentation it just reads finalized block, but particularly here previously `lastFinalizedBlock` method were actually triggering finalization process (which is odd), so just wanted to mention it. 
`But it will be removed with the new finalizer.` 😃